tools/c7n_mailer/c7n_mailer/slack_delivery.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import time
+
+import requests
+import six
+from c7n_mailer.email_delivery import EmailDelivery
+from c7n_mailer.ldap_lookup import Redis
+from c7n_mailer.utils import kms_decrypt, get_rendered_jinja
+from slackclient import SlackClient
+
+
+class SlackDelivery(object):
+
+    def __init__(self, config, session, logger):
+        if config.get('slack_token'):
+            config['slack_token'] = kms_decrypt(config, logger, session, 'slack_token')
+            self.client = SlackClient(config['slack_token'])
+        self.cache_engine = config.get('cache_engine', None)
+        if self.cache_engine == 'redis':
+            self.caching = Redis(redis_host=config.get('redis_host'),
+                                 redis_port=int(config.get('redis_port', 6379)), db=0)
+        self.config = config
+        self.logger = logger
+        self.session = session
+        self.email_handler = EmailDelivery(config, session, logger)
+
+    def get_to_addrs_slack_messages_map(self, sqs_message):
+        to_addrs_to_resources_map = self.email_handler.get_email_to_addrs_to_resources_map(sqs_message)
+        slack_messages = {}
+
+        # Check for Slack targets in 'to' action and render appropriate template.
+        for e in sqs_message.get('action', ()).get('to'):
+            if 'slack://owners' in e:
+                for to_addrs, resources in six.iteritems(to_addrs_to_resources_map):
+
+                    resolved_addrs = self.retrieve_user_im(list(to_addrs))
+
+                    if not resolved_addrs:
+                        continue
+
+                    for address, slack_target in resolved_addrs.iteritems():
+                        slack_messages[address] = get_rendered_jinja(slack_target, sqs_message, resources,
+                                                                     self.logger, 'slack_template', 'slack_default')
+                self.logger.debug("Sending to owners.")
```

Comment:
Could you add additional details in the log message here? Sending to owners is pretty vague-- no sense in logging something that won't be useful